test/useFormo.test-d.ts

```diff
@@ -1,37 +1,145 @@
-      expectType<string>(values.name);
-      expectType<string>(values.age);
-      return taskEither.left(values);
+    {
+      onSubmit: (values) => {
+        expectType<string>(values.name);
+        expectType<string>(values.age);
+        return Promise.resolve(failure(values));
+      },
+    }
+  );
+
+  expectType<string>(fieldProps("name").value);
+  expectType<(v: string) => unknown>(fieldProps("name").onChange);
+  expectType<NonEmptyArray<boolean> | undefined>(fieldProps("name").issues);
+
+  expectType<number>(fieldProps("age").value);
+  expectType<(v: number) => unknown>(fieldProps("age").onChange);
+  expectType<NonEmptyArray<boolean> | undefined>(fieldProps("age").issues);
+
+  expectType<Partial<Record<"name" | "age", NonEmptyArray<boolean>>>>(
+    fieldErrors
+  );
+}
+
+export function definedValidator() {
+  type Values = {
+    notValidated?: string;
+    name?: string;
+    surname?: string | null;
+    city: string | null;
+    address: string | null | undefined;
+  };
+  const initialValues: Values = {
+    city: null,
+    address: undefined,
+  };
+  useFormo(
+    {
+      initialValues,
+      // declaring _values here:
+      fieldValidators: (_values) => ({
+        name: validators.defined("required"),
+        surname: validators.defined("required"),
+        city: validators.defined("required"),
+        address: validators.defined("required"),
+      }),
     },
-  }
-);
+    {
+      onSubmit: (values) => {
+        expectType<string>(values.name);
+        expectType<string>(values.surname);
+        expectType<string>(values.city);
+        expectType<string>(values.address);
+        expectType<string | undefined>(values.notValidated);
 
-expectType<string>(fieldProps("name").value);
-expectType<(v: string) => unknown>(fieldProps("name").onChange);
-expectType<Option<NonEmptyArray<boolean>>>(fieldProps("name").issues);
+        return Promise.resolve(failure(null));
+      },
+    }
+  );
+}
 
-expectType<number>(fieldProps("age").value);
-expectType<(v: number) => unknown>(fieldProps("age").onChange);
-expectType<Option<NonEmptyArray<boolean>>>(fieldProps("age").issues);
+export function definedValidatorNoValuesParam() {
+  type Values = {
+    notValidated?: string;
+    name?: string;
+    surname?: string | null;
+    city: string | null;
+    address: string | null | undefined;
+  };
+  const initialValues: Values = {
+    city: null,
+    address: undefined,
+  };
+  useFormo(
+    {
+      initialValues,
+      // not declaring the values argument here:
+      fieldValidators: () => ({
+        name: validators.defined("required"),
+        surname: validators.defined("required"),
+        city: validators.defined("required"),
+        address: validators.defined("required"),
+      }),
+    },
+    {
+      onSubmit: (values) => {
+        expectType<string | undefined>(values.notValidated);
+        // all of the following end up being `unknown`, not sure how to fix this
```

Comment:
I tested by splitting `initialValues` and `fieldValidators` in separate input arguments for `useFormo`: it fixes this issues but introduce other ones